test/AutoDiff/compiler_crashers_fixed/sr15849-invalid-debug-info.swift

```diff
@@ -0,0 +1,139 @@
+// SR-15849: Mutating functions with control flow can cause assertion failure
+// for conflicting debug variable type.
+
+// RUN: %empty-directory(%t)
+// RUN: %target-build-swift -O -g -debug-info-format=dwarf -emit-module -module-name sr15849 -emit-module-path %t/sr15849.swiftmodule -swift-version 5 -c %S/sr15849-invalid-debug-info.swift
```

Comment:
Can you compile to LLVM IR or SIL add a few CHECK lines? Otherwise this test would pass even if swiftc gets symlinked to /bin/true :-)

Reply:
Yep. Not to SIL though. The assert is exposed only during LLVM IR codegen (in the debug info verifier)